app/Models/ComponentGroup.php

```diff
@@ -173,4 +174,23 @@ public function scopeVisible(Builder $query)
     {
         return $query->where('visible', self::VISIBLE_PUBLIC);
     }
+
+    /**
+     * Finds all used and visible component groups.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param \Illuminate\Support\Collection $usedComponentGroups
+     * @param bool $isAuthenticated
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeVisibleUsed(Builder $query, Collection $usedComponentGroups, $isAuthenticated)
```

Comment:
This seems to be doing a bit more than it should. This should, imo, be `scopeUsed`. Drop the auth check. The parent (caller) should then call `visible` if applicable.